src/components/Calculator.js

```diff
@@ -1,35 +1,46 @@
+      operation: null,
+      total: null,
+    };
+    this.clickHandler = this.clickHandler.bind(this);
+  }
+
+  clickHandler(e) {
+    this.setState((state) => calculate(state, e.target.textContent));
   }
 
   render() {
+    const { next, operation, total } = this.state;
     return (
       <div className="calculator">
-        <div className="display">0</div>
-        <button type="button" className="gray">AC</button>
-        <button type="button" className="gray">+/-</button>
-        <button type="button" className="gray">%</button>
-        <button type="button" className="orange">+</button>
-        <button type="button" className="gray">7</button>
-        <button type="button" className="gray">8</button>
-        <button type="button" className="gray">9</button>
-        <button type="button" className="orange">X</button>
-        <button type="button" className="gray">4</button>
-        <button type="button" className="gray">5</button>
-        <button type="button" className="gray">6</button>
-        <button type="button" className="orange">-</button>
-        <button type="button" className="gray">1</button>
-        <button type="button" className="gray">2</button>
-        <button type="button" className="gray">3</button>
-        <button type="button" className="orange">+</button>
-        <button type="button" className="gray zero">0</button>
-        <button type="button" className="gray">.</button>
-        <button type="button" className="orange">=</button>
+        <div className="display">{ next || operation || total || 0 }</div>
+        <button type="button" onClick={this.clickHandler} className="gray">AC</button>
+        <button type="button" onClick={this.clickHandler} className="gray">+/-</button>
+        <button type="button" onClick={this.clickHandler} className="gray">%</button>
+        <button type="button" onClick={this.clickHandler} className="orange">+</button>
+        <button type="button" onClick={this.clickHandler} className="gray">7</button>
+        <button type="button" onClick={this.clickHandler} className="gray">8</button>
+        <button type="button" onClick={this.clickHandler} className="gray">9</button>
+        <button type="button" onClick={this.clickHandler} className="orange">x</button>
+        <button type="button" onClick={this.clickHandler} className="gray">4</button>
+        <button type="button" onClick={this.clickHandler} className="gray">5</button>
+        <button type="button" onClick={this.clickHandler} className="gray">6</button>
+        <button type="button" onClick={this.clickHandler} className="orange">-</button>
+        <button type="button" onClick={this.clickHandler} className="gray">1</button>
+        <button type="button" onClick={this.clickHandler} className="gray">2</button>
+        <button type="button" onClick={this.clickHandler} className="gray">3</button>
+        <button type="button" onClick={this.clickHandler} className="orange">+</button>
```

Comment:
- [ ] Great job so far 👏🏻 However, we should be able to do all the math operations, in fact, you are missing the division operator and that's because you duplicated the addition operation, let's check it on line 28 and line 40, you are going to see that there are identical. Please let's correct that 👍🏻

![18](https://user-images.githubusercontent.com/79058364/188997948-ba4815bd-c489-45ba-b538-5ffa4c8123e5.png)

Reply:
Oh, you were right! They are so small that I couldn't see the difference between them, I already fixed it! Thank you for your review!